lighthouse-core/test/audits/treemap-data-test.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const TreemapData_ = require('../../audits/treemap-data.js');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
+const {loadSourceMapFixture, makeParamsOptional} = require('../test-utils.js');
+
+/* eslint-env jest */
+
+const TreemapData = {
+  audit: makeParamsOptional(TreemapData_.audit),
+  prepareTreemapNodes: makeParamsOptional(TreemapData_.prepareTreemapNodes),
+};
+
+/**
+ * @param {string} name
+ */
+function load(name) {
+  const data = loadSourceMapFixture(name);
+  if (!data.usage) throw new Error('exepcted usage');
+  return {...data, usage: data.usage};
+}
+
+/**
+ * @param {string} url
+ * @param {number} transferSize
+ * @param {LH.Crdp.Network.ResourceType} resourceType
+ */
+function generateRecord(url, transferSize, resourceType) {
+  return {url, transferSize, resourceType};
+}
+
+describe('TreemapData audit', () => {
+  describe('squoosh fixture', () => {
+    /** @type {import('../../audits/treemap-data.js').TreemapData} */
+    let treemapData;
+    beforeAll(async () => {
+      const context = {computedCache: new Map()};
+      const {map, content, usage} = load('squoosh');
+      const mainUrl = 'https://squoosh.app';
+      const scriptUrl = 'https://squoosh.app/main-app.js';
+      const networkRecords = [generateRecord(scriptUrl, content.length, 'Script')];
+
+      const artifacts = {
+        URL: {requestedUrl: mainUrl, finalUrl: mainUrl},
+        JsUsage: {[usage.url]: [usage]},
+        devtoolsLogs: {defaultPass: networkRecordsToDevtoolsLog(networkRecords)},
+        SourceMaps: [{scriptUrl: scriptUrl, map}],
+        ScriptElements: [{src: scriptUrl, content}],
+      };
+      const results = await TreemapData.audit(artifacts, context);
+
+      // @ts-expect-error: Debug data.
+      treemapData = results.details.treemapData;
+    });
+
+    it('basics', () => {
+      expect(Object.keys(treemapData)).toEqual(['scripts', 'resources']);
+    });
+
+    it('js', () => {
+      expect(treemapData.scripts).toMatchSnapshot();
+    });
+
+    it('resource summary', () => {
```

Comment:
Is it possible to move these to more specific tests? With a snapshot, when something changes all that's clear is that something has changed, and it may not be clear that it's a bad thing.

It also makes debugging start with a blank slate..instead of knowing that the failure is exactly that dependent modules aren't nested under their parents, or that names aren't trimmed of the sourceRoot, you have to start with, _something_ has changed, track down what that is and what caused it, then go read the file under test to figure out if the original intention was that dependent modules should be nested under their parents or that names are trimmed of the sourceRoot (and also are there cases when those things shouldn't happen...).

Reply:
I've made a few more specific tests.